### PR TITLE
Implement methods for additional fields, priority B

### LINKED ISSUE #12: New fields
Priority A (closed by #14):
- ~~024: e-manuscripta (instead of 856)~~
- ~~610 / 710: Körperschaften~~
- ~~300: Physische Beschreibung~~
- ~~500: Anmerkung~~
- ~~520: Inhaltsangabe~~
- ~~041: Sprache standardisiert~~

Priority B:
- 250: Entstehungstufe
- 264: original date (as string)
- 525: Begleitmaterial

Priority C:
- 581: Literatur

### PR DESCRIPTION
**Pull request by BalduinLandolt**
This PR adds methods for additional Marc fields.

Fields:  
- [x] 525: Anmerkung zu Begleitmaterial
- [x] 250: Entstehungsstufe
- [x] 264: Originaldatum (Transkription, wie es auf dem Brief steht)
- [x] 533/534/544: Verweis auf weitere Fassungen
- [x] 581: Literaturverweise

closes #12

### CODE REVIEW

File: alephmarcreader/abstractalephmarcreader.py [hunk 1]
```diff
@@ -353,4 +362,63 @@ def get_recipient_organisation(self):
...
+            date = self._handle_subfields_cardinality_max_one(self.__get_subfield_texts(field, 'c'))
+            place = self._handle_subfields_cardinality_max_one(self.__get_subfield_texts(field, 'a'))
+            # TODO Ignore Fields starting wit '['?
+
+            orig_date = self.OriginalDate(date, place)
+            original_dates.append(orig_date)
+
+        return original_dates
+
+    get_document_state.__annotations__ = {'return': [OriginalDate]}
```
Comment (tobiasschweizer) [inline alephmarcreader/abstractalephmarcreader.py]: should be `get_original_date `

Comment (BalduinLandolt) [reply alephmarcreader/abstractalephmarcreader.py]: aye. I'll fix that

**Comment by BalduinLandolt** [comment]
@tobiasschweizer I think that's it. Please review! :)

File: alephmarcreader/abstractalephmarcreader.py [hunk 2]
```diff
@@ -353,4 +362,117 @@ def get_recipient_organisation(self):
 
         return recipient_organisation
 
-    get_recipient_organisation.__annotations__ = {'return': [Organisation]}
\ No newline at end of file
+    get_recipient_organisation.__annotations__ = {'return': [Organisation]}
+
+    def get_supplement_remarks(self):
+        """
+        Returns comments on supplementing material following the letter. (See also Description.supplement.)
```
Comment (tobiasschweizer) [inline alephmarcreader/abstractalephmarcreader.py]: How is this related to `Description.supplement`?

File: alephmarcreader/abstractalephmarcreader.py [hunk 3]
```diff
@@ -353,4 +362,117 @@ def get_recipient_organisation(self):
...
+
+    def get_document_state(self):
+        """
+        Returns a description of the document state. (i.e. "original", "copy", "draft", etc.)
+        :return: [str]
+        """
+        states = []
+
+        for field in self.__get_field('250'):
+            footnote_text = self._handle_subfields_cardinality_max_one(self.__get_subfield_texts(field, 'a'))
```
Comment (tobiasschweizer) [inline alephmarcreader/abstractalephmarcreader.py]: You could call this variable something else, just to make it clear.

File: alephmarcreader/abstractalephmarcreader.py [hunk 4]
```diff
@@ -353,4 +362,117 @@ def get_recipient_organisation(self):
...
+    get_original_date_and_place.__annotations__ = {'return': [OriginalDate]}
+
+    def get_references_to_related_entries(self):
+        """
+        Returns a list of related entries.
+        That is usually the case, if there are several document stages of the same letter.
+        (E.g. Original, draft, copy, etc.)
+        :return: [str]
+        """
+        references = []
```
Comment (tobiasschweizer) [inline alephmarcreader/abstractalephmarcreader.py]: So this combines subfields from different fields into one array?

File: alephmarcreader/abstractalephmarcreader.py [hunk 5]
```diff
@@ -353,4 +362,117 @@ def get_recipient_organisation(self):
...
+
+    def get_bibliographic_references(self):
+        """
+        Returns bibliographic references relevant to the letter. (Usually, where it has been printed.)
+        :return: [str]
+        """
+        references = []
+
+        for field in self.__get_field('581'):
+            ref = u''
```
Comment (tobiasschweizer) [inline alephmarcreader/abstractalephmarcreader.py]: I don't understand what this prefix means.

File: alephmarcreader/abstractalephmarcreader.py (hunk 2 above)
Comment (BalduinLandolt) [reply alephmarcreader/abstractalephmarcreader.py]: Description.supplement (or Field 300, Subfield e) lists all "Begleitmaterial", the supplement remarks (Field 525) explains, what the Begleitmaterial is.

I was hoping we could get rid of one of these things, but aparently they're quite different. E.g.: 000054780 
- 300e could be something like `Beilage (3 Bl. (4, 4a, 4b))`
- 525 could be something like `Dem Brief liegt als fo. 4 (24 x 19 cm), fo. 4a (23,5 x 18,5 cm) und fo. 4b (12 x 18,5 cm) eine autographe Frühfassung von Daniel Bernoullis Abhandlung "Specimen theoriae novae de mensura sortis" bei. Diese unterscheidet sich von der Druckfassung in: CP V (1730/31) pp.175-192 (St.22).`

File: alephmarcreader/abstractalephmarcreader.py (hunk 3 above)
Comment (BalduinLandolt) [reply alephmarcreader/abstractalephmarcreader.py]: yes... that's copy-paste

File: alephmarcreader/abstractalephmarcreader.py (hunk 4 above)
Comment (BalduinLandolt) [reply alephmarcreader/abstractalephmarcreader.py]: Yes.

I could not quite make sense of the Fields 533, 534, 544 (probably they also don't really do make sense in the first place).  
But from what I can tell, the general idea is that different stages of a letter refer to other states from the same letter in different fields. (E.g. the copy would refer to its original in a different field than the original to its copy.)

Sulamith and I aggreed that it would be the easiest to just combine these things on BEOL, as it's basically one type of information, that HAN required to splitt up.  
(Also, currently, these fields are not used systematically. But that should change at some point.)

If you don't like this, feel free to make a suggestion.

File: alephmarcreader/abstractalephmarcreader.py (hunk 5 above)
Comment (BalduinLandolt) [reply alephmarcreader/abstractalephmarcreader.py]: Subfield i contains information on what subfield a is.  
It can be things like "Druck" or "Inhaltsangabe bei" and stuff.  
a should be the the bibliographic reference itself (though that's not always the case, we also have cases, where i is missing and a is "Druck in: bla bla").

So basically, I consider a the body of the information, because tht's what counts. But in case we have Subfield i, I add this as a prefix to the core information, followed by a colon and a whitespace. (`[PREFIX: ]BODY` or `[SUBFIELD_I: ]SUBFIELD_A`,  
e.g. :000056870  
`Druck: Joh. I B. Briefe 1, p.444` for `<datafield tag="581" ind1=" " ind2=" ">
    <subfield code="i">Druck</subfield>
    <subfield code="a">Joh. I B. Briefe 1, p.444</subfield>
  </datafield>`)

To me, that seemed like a reasonable way to handle this, for now. And as soon as the propper bibliographies on BEOL are complete, this field should be obsolete anyway.  
But feel free to suggest a better solution.

File: alephmarcreader/abstractalephmarcreader.py (hunk 2 above)
Comment (tobiasschweizer) [reply alephmarcreader/abstractalephmarcreader.py]: I was just confused to find a reference to a type that is not returned by this method.

So if these fields are semantically related, they should probably be combined into one method. Otherwise, keep them separate but do not cross reference.

File: alephmarcreader/abstractalephmarcreader.py (hunk 4 above)
Comment (tobiasschweizer) [reply alephmarcreader/abstractalephmarcreader.py]: I think that's ok how it is.

File: alephmarcreader/abstractalephmarcreader.py (hunk 5 above)
Comment (tobiasschweizer) [reply alephmarcreader/abstractalephmarcreader.py]: I was just thinking that if you need this information later, you would have to parse the string which would be cumbersome. You could return an instance of the class with separate members.

File: alephmarcreader/abstractalephmarcreader.py (hunk 3 above)
Comment (tobiasschweizer) [reply alephmarcreader/abstractalephmarcreader.py]: ok :-)

File: alephmarcreader/abstractalephmarcreader.py (hunk 1 above)
Comment (tobiasschweizer) [reply alephmarcreader/abstractalephmarcreader.py]: ok

File: alephmarcreader/abstractalephmarcreader.py (hunk 2 above)
Comment (BalduinLandolt) [reply alephmarcreader/abstractalephmarcreader.py]: They are semantically related in the sense that they both refer to the supplement; but 300e goes better with physical description than with 525.  
I'll remove the reference then.

File: alephmarcreader/abstractalephmarcreader.py (hunk 4 above)
Comment (BalduinLandolt) [reply alephmarcreader/abstractalephmarcreader.py]: Alternatively, I could create a subclass that stores an array for these fields separately, and has a method that returns a combined array of all three fields.  
Might come in handy, if ever we need them separately anyway.

If you like the idea, just assign an issue to me.

File: alephmarcreader/abstractalephmarcreader.py (hunk 5 above)
Comment (BalduinLandolt) [reply alephmarcreader/abstractalephmarcreader.py]: ok, will do. In that case, I just add a method `get_pretty_string()` that returns them combined.

Comment (BalduinLandolt) [reply alephmarcreader/abstractalephmarcreader.py]: is that better, now?

File: alephmarcreader/abstractalephmarcreader.py (hunk 2 above)
Comment (tobiasschweizer) [reply alephmarcreader/abstractalephmarcreader.py]: Ok

File: alephmarcreader/abstractalephmarcreader.py (hunk 4 above)
Comment (tobiasschweizer) [reply alephmarcreader/abstractalephmarcreader.py]: I think the current implementation is sufficient.

File: alephmarcreader/abstractalephmarcreader.py (hunk 5 above)
Comment (tobiasschweizer) [reply alephmarcreader/abstractalephmarcreader.py]: That's nice. Could you test for the members `prefix` and `reference` in the test, too?

Comment (BalduinLandolt) [reply alephmarcreader/abstractalephmarcreader.py]: voila :)

**Comment by BalduinLandolt** [comment]
is it time for merging now?
